pytorch_lightning/core/lightning.py

```diff
@@ -179,6 +179,12 @@ def logger(self):
         """ Reference to the logger object in the Trainer. """
         return self.trainer.logger if self.trainer else None
 
+    def _apply_batch_transfer_handler(self, batch: Any, device: Optional[torch.device] = None):
+        batch = self.on_before_batch_transfer(batch)
+        batch = self.transfer_batch_to_device(batch, device)
+        batch = self.on_after_batch_transfer(batch)
+        return batch
+
```

Comment:
maybe one can find a better place where to place this method :)

Reply:
how about DataHooks? or do you have any other suggestions?

Reply:
I meant actually just the place within the file. There is no real logic in the ordering of methods in this file, so it's hard to decide where a good place is :) 
Maybe place it close to where we have transfer_batch_to_device?

Reply:
`transfer_batch_to_device` is in `DataHooks` and putting it there won't be right since it's a `LightningModule` method and `DataHooks` is inherited by `LightningDataModule` too. I'll see if I can find a better place for it.